plugins/vault-backend/src/service/vaultApi.ts

```diff
@@ -88,11 +93,58 @@ export interface VaultApi {
 export class VaultClient implements VaultApi {
   private vaultConfig: VaultConfig;
   private readonly limit = plimit(5);
+  private token: string = '';
 
   constructor(options: { config: Config }) {
     this.vaultConfig = getVaultConfig(options.config);
   }
 
+  static async fromConfig(config: Config): Promise<VaultApi> {
+    const vaultClient = new VaultClient({ config });
+
+    if (typeof vaultClient.vaultConfig.token === 'string') {
+      vaultClient.token = vaultClient.vaultConfig.token;
+    } else {
+      vaultClient.token = await vaultClient.kubernetesLogin(
+        vaultClient.vaultConfig.baseUrl,
+        vaultClient.vaultConfig.token,
+      );
+    }
+
+    return vaultClient;
+  }
+
+  private async kubernetesLogin(
+    baseUrl: string,
+    kubernetesCfg: VaultKubernetesAuthConfig,
+  ): Promise<string> {
+    const jwt = readFileSync(
+      kubernetesCfg.serviceAccountTokenPath ||
+        '/var/run/secrets/kubernetes.io/serviceaccount/token',
+      'utf-8',
+    );
```

Comment:
using a sync method will block the event loop, meaning that no operation will be executed until reading has completed. It would be better to switch to an async method. We could also avoid including `fs-extra` as a dependency, as an almost identical method is included in node.

`import { readFile } from 'fs/promise';` 


```suggestion
    const jwt = await readFile(
      kubernetesCfg.serviceAccountTokenPath ||
        '/var/run/secrets/kubernetes.io/serviceaccount/token',
      'utf-8',
    );
```

Reply:
Replaced. I used the `fs-extra` because I saw it in other places as well. But much cleaner if we use the included by by node